test/gtest/meson.build

```diff
@@ -0,0 +1,17 @@
+gtest_src = [
+  'src/gtest-all.cc',
+  'src/gtest.cc',
+  'src/gtest-death-test.cc',
+  'src/gtest-filepath.cc',
+  'src/gtest_main.cc',
+  'src/gtest-port.cc',
+  'src/gtest-printers.cc',
+  'src/gtest-test-part.cc',
+  'src/gtest-typed-test.cc',
+]
+gtest_inc = include_directories('src', 'include')
+
+gtest = static_library('gtest', gtest_src,
+                       include_directories: gtest_inc)
+gtest_dep = declare_dependency(link_with: gtest,
+                               include_directories: gtest_inc)
```

Comment:
The gtest directory is a copy of upstream googletest release.  Thus, we'd like to avoid additions/modifications at this level of the directory tree as much as possible.  Please incorporate this part into test/meson.build instead.  Also, please make test/meson.build equivalent to the test/Makefile.am configuration (e.g. compilation flags, etc).